include/h2o.h

```diff
@@ -919,6 +890,18 @@ typedef struct st_h2o_conn_callbacks_t {
      * debug state callback (optional)
      */
     h2o_http2_debug_state_t *(*get_debug_state)(h2o_req_t *req, int hpack_enabled);
+    /**
+     * returns number of closed idle connections
+     */
+    int (*close_idle_connection)(h2o_conn_t *conn);
+    /**
+     * shutdown of connection is requested (optional)
+     */
+    void (*request_shutdown)(h2o_conn_t *conn);
+    /**
+     * for each request (optional)
```

Comment:
I think this callback is currently mandatory, isn't it? We use it in the status handler as you correctly noticed.

Reply:
Yes you're right, I refactored to check for NULL when previously this was not the case.